x/gov/types/tally_router.go

```diff
@@ -0,0 +1,62 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ TallyRouter = (*tallyrouter)(nil)
+
+// TallyRouter implements a governance Tally router.
+//
+// TODO: Use generic router (ref #3976).
+type TallyRouter interface {
+	AddRoute(r string, h TallyStrategy) (rtr TallyRouter)
+	HasRoute(r string) bool
+	GetRoute(path string) (h TallyStrategy)
+}
+
+type tallyrouter struct {
+	routes map[string]TallyStrategy
+	sealed bool
+}
+
+// NewTallyRouter creates a new Router interface instance
+func NewTallyRouter() TallyRouter {
+	return &tallyrouter{
+		routes: make(map[string]TallyStrategy),
+	}
+}
+
+// AddRoute adds a governance handler for a given path. It returns the Router
+// so AddRoute calls can be linked. It will panic if the router is sealed.
+func (rtr *tallyrouter) AddRoute(path string, t TallyStrategy) TallyRouter {
+	if rtr.sealed {
```

Comment:
Where sealed is set?

Reply:
Ah, I actually meant to get rid of that.  I got rid of the .Seal() function on this router, because the point is we want other modules to add Routes dynamically.

Reply:
You can remove this check and the sealed variable from tally_router